### PR TITLE
feat: update manifest parameters to accept YAML string or file path

### PR DESCRIPTION
**Pull request by aaronsteers**
# fix: revert parameter names from manifest_input to manifest

## Summary

This PR reverts parameter names from `manifest_input` back to `manifest` across all affected functions in the connector builder MCP server, as requested by @aaronsteers in PR feedback. The changes maintain the string input functionality that accepts either raw YAML content or file paths while restoring the original parameter naming convention.

**Key Changes:**
- Reverted parameter names in `validate_manifest()`, `execute_stream_test_read()`, `get_resolved_manifest()`, and `populate_dotenv_missing_secrets_stubs()`
- Updated function signatures, docstrings, and internal variable references
- Fixed OSError handling in `parse_manifest_input()` to properly catch "File name too long" errors when YAML strings are mistaken for file paths
- Updated test function calls and assertions to use the reverted parameter names

The core auto-detection logic that distinguishes between YAML strings and file paths remains unchanged, preserving the context window optimization benefits for LLMs.

## Review & Testing Checklist for Human

- [ ] **Test auto-detection with edge cases**: Verify the system correctly handles very long YAML strings, file paths that contain YAML-like content, and unusual file formats
- [ ] **Verify parameter consistency**: Check that all function calls across the codebase use the correct `manifest` parameter name (no lingering `manifest_input` references)
- [ ] **Test error handling scenarios**: Confirm that file system errors are properly reported and that the OSError catch doesn't mask legitimate issues
- [ ] **End-to-end workflow validation**: Test the complete manifest validation, stream testing, and secrets management workflows with both YAML strings and file paths

**Recommended Test Plan:**
1. Test with a real YAML file path to ensure file loading works
2. Test with raw YAML string content to ensure parsing works  
3. Test with invalid file paths and malformed YAML to verify error handling
4. Test the secrets management functions with manifest-based auto-detection

---

### Diagram

```mermaid
%%{ init : { "theme" : "default" }}%%
graph TD
    A["connector_builder_mcp/_connector_builder.py"]:::major-edit
    B["connector_builder_mcp/_secrets.py"]:::major-edit
    C["connector_builder_mcp/_util.py"]:::major-edit
    D["tests/test_secrets.py"]:::minor-edit
    E["tests/test_integration.py"]:::minor-edit
    F["parse_manifest_input()"]:::context
    
    A -->|calls| F
    B -->|calls| F
    C -->|contains| F
    D -->|tests| B
    E -->|tests| A
    
    subgraph Legend
        L1[Major Edit]:::major-edit
        L2[Minor Edit]:::minor-edit
        L3[Context/No Edit]:::context
    end

classDef major-edit fill:#90EE90
classDef minor-edit fill:#87CEEB  
classDef context fill:#FFFFFF
```

### Notes

- All local tests pass (63/63) and CI checks are green
- The OSError handling fix was critical to resolve the "File name too long" error that occurred when trying to create Path objects from YAML string content
- This change addresses specific feedback from @aaronsteers while maintaining the improved API functionality
- Session: https://app.devin.ai/sessions/ba10520228444842884d3101bcb49827
- Requested by: @aaronsteers (AJ Steers)

> [!IMPORTANT]
> **Auto-merge enabled.**
> 
> _This PR is set to merge automatically when all requirements are met._

### CODE REVIEW

File: tests/test_secrets.py [hunk 1]
```diff
@@ -205,9 +208,12 @@ def test_populate_dotenv_missing_secrets_stubs_combined_mode():
             }
         }
 
+        import yaml
```
Comment (aaronsteers) [inline tests/test_secrets.py:211]: Move import to top of file please.

File: tests/test_secrets.py [hunk 2]
```diff
@@ -245,7 +251,10 @@ def test_populate_dotenv_missing_secrets_stubs_empty_manifest():
             }
         }
 
-        result = populate_dotenv_missing_secrets_stubs(f.name, manifest=manifest)
+        import yaml
```
Comment (aaronsteers) [inline tests/test_secrets.py:254]: Ditto here and any other yaml imputs.
